packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -207,6 +207,39 @@ mixin class ChangeNotifier implements Listenable {
   @protected
   bool get hasListeners => _count > 0;
 
+  /// Dispatches event of object creation to [MemoryAllocations.instance].
+  ///
+  /// If the event was already dispatched or [kFlutterMemoryAllocationsEnabled]
+  /// is false, the method is noop.
+  ///
+  /// Tools like leak_tracker use the event of object creation to help
+  /// developers identify the owner of the object, for troubleshooting purposes,
+  /// by taking stack trace at the moment of the event.
+  ///
+  /// But, as [ChangeNotifier] is mixin, it does not have its own constructor. So, it
+  /// communicates object creation in first `addListener`, that results
+  /// in the stack trace pointing to `addListener`, not to constructor.
+  ///
+  /// To make debugging easier, invoke [ChangeNotifier.maybeDispatchObjectCreation]
+  /// in constructor of the class. It will help
+  /// to identify the owner.
+  ///
+  /// Make sure to invoke it with condition `if (kFlutterMemoryAllocationsEnabled) ...`
+  /// so that the method is tree-shaken away when the flag is false.
+  @protected
+  void maybeDispatchObjectCreation() {
```

Comment:
Adding a new public method to ChangeNotifier is a breaking change for all users that implement ChangeNotifier

```
class ChangeNotifier2 implements ChangeNotifier {
  // ...
}
```

Last year, there was a similar [breaking change](https://github.com/flutter/flutter/commit/16b7348155c327a87d36570b2a1532c7b3ecfd70), which was converted to a static method instead to make it non-breaking.

https://github.com/flutter/flutter/blame/e1428bffd464082e8853dcfa54756be0812a900e/packages/flutter/lib/src/foundation/change_notifier.dart#L173C3-L176

Can we do this here, too?

Reply:
@passsy You're right, we missed this in our review. It would be great if you (or whoever else suffers under this) could donate some tests to https://github.com/flutter/tests so next time somebody attempts to add a method here tests are failing to remind us that we shouldn't add new methods here.

@polina-c Can you please look into how we can refactor this without causing pain to people that implement ChangeNotifier?

Reply:
https://github.com/flutter/flutter/pull/134953